File: addon-cache/src/main/java/io/nerv/cache/condition/RedisCacheCondition.java
package io.nerv.cache.condition;

import cn.hutool.core.util.StrUtil;
import org.springframework.context.annotation.Condition;
import org.springframework.context.annotation.ConditionContext;
import org.springframework.core.type.AnnotatedTypeMetadata;

/**
 * 缓存启用条件， 是否启用redis
 * @author: S.PKAQ
 * @Datetime: 2018/9/28 12:38
 */
public class RedisCacheCondition implements Condition {
    @Override
    public boolean matches(ConditionContext context, AnnotatedTypeMetadata metadata) {
        String cacheType = context.getEnvironment().getProperty("eva.cache.type");
        return StrUtil.isBlank(cacheType) || "redis".equals(cacheType);
    }
}

File: eva-core/src/main/kotlin/io/nerv/core/cache/condition/RedisCacheCondition.kt
package io.nerv.core.cache.condition

import org.springframework.context.annotation.Condition
import org.springframework.context.annotation.ConditionContext
import org.springframework.core.type.AnnotatedTypeMetadata

/**
 * 缓存启用条件， 是否启用redis
 * @author: S.PKAQ
 * @Datetime: 2018/9/28 12:38
 */
class RedisCacheCondition : Condition {
    override fun matches(context: ConditionContext, metadata: AnnotatedTypeMetadata): Boolean {
        val cacheType = context.environment.getProperty("eva.cache.type")
        return "redis" == cacheType
    }
}